contribs/gmf/src/directives/timeslider.js

```diff
@@ -0,0 +1,182 @@
+          var otherHandle = $(this).siblings('.ui-slider-handle');
+          if (otherHandle.length > 0) {
+            //test if other handle has its toolip hidden (do not call show for nothing)
+            if (!$(otherHandle[0]).attr('aria-describedby')) {
+              otherHandle.tooltip('show');
+            }
+          }
+        });
+
+        element.find('.ui-slider-handle').on('hide.bs.tooltip', function(jqEvt) {
+          if (element.find(this).length > 0) {
+            jqEvt.preventDefault();
+          }
+        });
+      }
+
+      function updateTooltip_(e, slider) {
+        var handle = $(slider.handle);
+        var left  = handle.position().left;
+        var tooltip = $('#' + handle.attr('aria-describedby'));
+        var dateText  = $filter('date')(new Date(slider.value), dateFormat);
+        handle.attr('data-original-title', dateText).tooltip('fixTitle');
+        tooltip.css('left', left - tooltip.width() / 2);
+        tooltip.find('.tooltip-inner').text(dateText);
+      }
+
+      function initToolTip_(i) {
+        var el = $(this);
+        if (!el.attr('aria-describedby')) {
+          var dateText  = Array.isArray(ctrl.dates) ?
+          $filter('date')(new Date(ctrl.dates[i]), dateFormat) :
+          $filter('date')(new Date(ctrl.dates), dateFormat);
+
+          el.tooltip({
+            'selector': 'span.ui-slider-handle',
+            'trigger': 'manual',
+            'placement' : i === 0 && 'bottom' || 'top',
+            'title' : dateText
+          })
+          .tooltip('show');
+        }
+      }
+
+      function onSliderReleased_(e, sliderUi) {
+        var sDate, eDate;
+        updateTooltip_(e, sliderUi);
+        if (sliderUi.values) {
+          sDate = new Date(sliderUi.values[0]);
+          eDate = new Date(sliderUi.values[1]);
+        } else {
+          sDate = new Date(sliderUi.value);
+        }
+
+        ctrl.onDateSelected({
+          time : {
+            start : sDate,
+            end : eDate
+          }
+        });
+      }
+    }
+  };
+};
+
+gmf.module.directive('gmfTimeSlider', gmf.TimeSlider);
+
+
+/**
+ * TimeSliderController - directive conttroller
+ * @param {!angular.Scope} $scope Angular scope.
+ * @constructor
+ * @export
+ * @ngInject
+ * @ngdoc controller
+ * @ngname gmfTimeSliderController
+ */
+gmf.TimeSliderController = function($scope) {
+
+
+  /**
+   * @function
+   * @export
+   */
+  this.onDateSelected;
+
+
+  /**
+   * @type {Object}
+   * @export
+   */
+  this.time;
+
+
+  var min = new Date(this.time.minValue).getTime();
+  var max = new Date(this.time.maxValue).getTime();
+  var step = (max - min) / (1000 * 3600 * 24);
+
+  /**
+   * @type {Object}
+   * @export
+   */
+  this.sliderOptions = {
+    'range' : this.time.mode === 'range',
+    'min' : min,
+    'max' : max,
+    'step' : step
+  };
+
+  /**
+   * @type {Array|number}
+   * @export
+   */
+  this.dates = this.time.mode === 'range' && [min, max] || Math.ceil((max - min) / 2);
```

Comment:
missing min:
Math.ceil((max - min) / 2)
should be Math.ceil(min + (max - min) / 2)